crates/rattler_lock/src/parse/deserialize.rs

```diff
@@ -219,37 +220,89 @@ fn parse_from_lock<P>(
+                                                {
+                                                    let merged_package = Cow::Borrowed(
+                                                        &conda_packages[first_package_idx],
+                                                    );
+                                                    let package = matching_packages.fold(
+                                                        merged_package,
+                                                        |acc, &next_package_idx| {
+                                                            if let Cow::Owned(merged) = acc.merge(
+                                                                &conda_packages[next_package_idx],
+                                                            ) {
+                                                                Cow::Owned(merged)
+                                                            } else {
+                                                                acc
+                                                            }
+                                                        },
+                                                    );
+                                                    Some(match package {
+                                                        Cow::Borrowed(_) => first_package_idx,
+                                                        Cow::Owned(package) => {
+                                                            conda_packages.push(package);
+                                                            conda_packages.len() - 1
+                                                        }
+                                                    })
+                                                } else {
+                                                    None
+                                                }
                                             } else {
-                                                all_packages.iter().find(|&idx| {
-                                                    let conda_package = &conda_packages[*idx];
-                                                    name.as_ref().map_or(true, |name| {
-                                                        name == &conda_package.record().name
-                                                    }) && version.as_ref().map_or(true, |version| {
-                                                        version == &conda_package.record().version
-                                                    }) && build.as_ref().map_or(true, |build| {
-                                                        build == &conda_package.record().build
-                                                    }) && subdir.as_ref().map_or(true, |subdir| {
-                                                        subdir == &conda_package.record().subdir
+                                                all_packages
```

Comment:
small request: could we please add a comment what this part of branching is doing?